imap_processing/codice/codice_l1a.py

```diff
@@ -391,45 +401,26 @@ def reshape_data(self) -> None:
         3D arrays representing dimensions such as spin sectors, positions, and
         energies (depending on the data product).
         """
+        # This will contain the reshaped data for all counters
         self.data = []
 
-        # For CoDICE-lo, data are a 3D arrays with a shape representing
-        # [<num_positions>,<num_spin_sectors>,<num_energy_steps>]
-        if self.config["instrument"] == "lo":
-            for packet_data in self.raw_data:
-                if packet_data:
-                    reshaped_packet_data = np.array(
-                        packet_data, dtype=np.uint32
-                    ).reshape(
-                        (
-                            self.config["num_counters"],
-                            self.config["num_energy_steps"],
-                            self.config["num_positions"],
-                            self.config["num_spin_sectors"],
-                        )
-                    )
-                    self.data.append(reshaped_packet_data)
-                else:
-                    self.data.append(None)
-
-        # For CoDICE-hi, data are a 3D array with a shape representing
-        # [<num_energy_steps>,<num_positions>,<num_spin_sectors>]
-        elif self.config["instrument"] == "hi":
-            for packet_data in self.raw_data:
-                if packet_data:
-                    reshaped_packet_data = np.array(
-                        packet_data, dtype=np.uint32
-                    ).reshape(
-                        (
-                            self.config["num_counters"],
-                            self.config["num_energy_steps"],
-                            self.config["num_positions"],
-                            self.config["num_spin_sectors"],
-                        )
-                    )
-                    self.data.append(reshaped_packet_data)
-                else:
-                    self.data.append(None)
+        # Typically, data are a 4D arrays with a shape representing some
+        # combination of <num_counters>, <num_positions>, <num_spin_sectors>,
+        # and <num_energy_steps>. However, the existence and order of these
+        # dimensions can vary depending on the specific data product, so we
+        # define this in the "dims" value configuration dictionary. The number
+        # of counters is the first dimension/axis.
+        reshape_dims = (self.config["num_counters"], *self.config["dims"].values())
+
+        # For each packet/epoch, reshape the data along these dimensions
+        for packet_data in self.raw_data:
+            if packet_data:
+                reshaped_packet_data = np.array(packet_data, dtype=np.uint32).reshape(
+                    reshape_dims
+                )
+                self.data.append(reshaped_packet_data)
+            else:
+                self.data.append(None)
```

Comment:
Do you want `None` here? If there is no data in the packet is that worth raising or just passing here to continue on?

Reply:
I now realize that these conditional statements are leftover from several months ago when I was dealing with decompression issues in the input data, which have since been resolved. So I think this can be removed altogether.